src/main/java/co/rsk/federate/signing/hsm/message/PowHSMSignerMessage.java

```diff
@@ -77,6 +115,59 @@ public Sha256Hash getSigHash() {
         return sigHash;
     }
 
+    public JsonNode getMessageToSign(int version) {
+        ObjectNode messageToSend = new ObjectMapper().createObjectNode();
+        messageToSend.put(TX.getFieldName(), getBtcTransactionSerialized());
+        messageToSend.put(INPUT.getFieldName(), inputIndex);
+        if (version == HSM_VERSION_5) {
+            if (hasWitness()) {
+                populateWithSegwitValues(messageToSend);
+            } else {
+                populateWithLegacyValues(messageToSend);
+            }
```

Comment:
Maybe move this into its own function?

Reply:
```
public JsonNode getMessageToSign(int version) {
        ObjectNode messageToSend = new ObjectMapper().createObjectNode();
        messageToSend.put(TX.getFieldName(), getBtcTransactionSerialized());
        messageToSend.put(INPUT.getFieldName(), inputIndex);
        if (version == HSM_VERSION_5) {
            populateWithHsmVersion5Values(messageToSend);
        }
        return messageToSend;
    }

    private void populateWithHsmVersion5Values(ObjectNode messageToSend) {
        if (hasWitness()) {
            populateWithSegwitValues(messageToSend);
        } else {
            populateWithLegacyValues(messageToSend);
        }
    }
``` 

Do you mean like this?

There is no prob for me in doing this change, but I don't see much benefit. Also, we agreed that soon, this is supposed to be refactored in benefit of use polymorphism. So I think it is better to focus the next effort on that way.